apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Snippets/BigtableClientSnippets.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Bigtable.V2.IntegrationTests;
+using Google.Cloud.ClientTesting;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.Snippets
+{
+    [SnippetOutputCollector]
+    [Collection(nameof(BigtableClientSnippetsFixture))]
+    public class BigtableClientSnippets
+    {
+        private readonly BigtableClientSnippetsFixture _fixture;
+
+        public BigtableClientSnippets(BigtableClientSnippetsFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public async Task CheckAndMutateRowAsync()
+        {
+            var client = _fixture.TableClient;
+            var projectId = _fixture.TableName.ProjectId;
+            var instanceId = _fixture.TableName.InstanceId;
+            var tableId = _fixture.TableName.TableId;
+
+            await client.MutateRowAsync(
+                new TableName(projectId, instanceId, tableId),
+                "user12345",
+                Mutations.SetCell("Address", "Country", "USA"));
+
+            // Snippet: CheckAndMutateRowAsync(TableName,BigtableByteString,RowFilter,*)
+            CheckAndMutateRowResponse response = await client.CheckAndMutateRowAsync(
+                new TableName(projectId, instanceId, tableId),
+                "user12345",
+                // If the latest value in the "Address:Country" column is "USA"...
+                RowFilters.Chain(
+                    RowFilters.FamilyNameRegex("Address"),
+                    RowFilters.ColumnQualifierRegex("Country"),
+                    RowFilters.CellsPerColumnLimit(1),
+                    RowFilters.ValueRegex("USA")),
```

Comment:
Is ValueRegex the most appropriate method to use even for simple literal matches?

Reply:
Yes. I guess we could offer a `RowFilters.Value(string)` method which delegates to `ValueRegex` with an escaped string (and similarly for the other `...Regex` methods). Does that sound like a good idea? If so I can followup with another PR.

Reply:
Let's create a feature request issue, and check with the Bigtable team. It sounds like something that should probably be consistent across languages.

Reply:
Ok, will do.